File: go/difference-of-squares/difference_of_squares.go
package diffsquares

// SquareOfSum returns the square of the sum of n numbers.
func SquareOfSum(n int) int {
	sum := (n*n + n) / 2
	return sum * sum
}

// SumOfSquares returns the sum of the square of n numbers.
func SumOfSquares(n int) int {
	return (n * (n + 1) * (2*n + 1)) / 6
}

// Difference returns the square of the sum of n numbers.
func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}

File: c/difference-of-squares/src/difference_of_squares.c
#include "difference_of_squares.h"

int square_of_sum(int n)
{
  // n(n + 1) / 2
  int sum = n * (n + 1) / 2;
  return sum * sum;
}

int sum_of_squares(int n)
{
  // n(n+1)(2n+1)/6
  return n * (n + 1) * (2 * n + 1) / 6;
}

int difference_of_squares(int n)
{
  return square_of_sum(n) - sum_of_squares(n); 
}
